src/screens/Profiles/MediaPages/index.js

```diff
@@ -120,40 +133,146 @@ class MediaPages extends PureComponent {
     }
   }
 
+  onLibraryEntryUpdated = (data) => {
+    // Check to see if we got this event from something other than 'quickupdate'
+    const { id, newEntry, source } = data;
+    const { libraryEntry } = this.state;
+    if (!newEntry || source === 'mediapage') return;
+
+    if (libraryEntry && libraryEntry.id == id) {
+      this.setState({ libraryEntry: newEntry });
+    }
+  }
+
+  onLibraryEntryDeleted = (data) => {
+    // Check to see if we got this event from something other than 'quickupdate'
+    const { id, source } = data;
+    const { libraryEntry } = this.state;
+    if (source === 'mediapage') return;
+
+    if (libraryEntry && libraryEntry.id == id) {
+      this.setState({ libraryEntry: null });
+    }
+  }
+
+  onEpisodeProgress = async (number) => {
+    const { media, libraryEntry } = this.state;
+    if (!media) return;
+
+    let changes = {};
+
+    // The media count
+    const mediaCount = media.episodeCount || media.chapterCount;
+
+    // Check progress is within the bounds
+    let progress = number;
+    if (progress < 0) progress = 0;
+    if (mediaCount && progress > mediaCount) {
+      progress = mediaCount;
+    }
+    changes = { ...changes, progress };
+
+    // Mark entry as completed if the media has finished airing and the progress is the same as the count.
```

Comment:
We might want to not check the media status here, Could hit an issue where DB might be slower to update than the final episode to air.

Reply:
hmm, what should i change the logic to? just check if progress is the same as the count?

Reply:
Yeah, that's all web does.